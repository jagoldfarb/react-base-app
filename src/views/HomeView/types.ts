export interface IHomeViewProps {
  name: string;
}
