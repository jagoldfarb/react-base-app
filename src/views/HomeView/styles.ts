import styled from 'styled-components';

import { COLORS } from 'helpers/enums/colors';

export const HomeContainer = styled.div`
  background-color: ${COLORS.white};
`;
