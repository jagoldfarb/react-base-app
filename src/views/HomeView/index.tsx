import React, { FunctionComponent } from 'react';

import { IHomeViewProps } from './types';
import { HomeContainer } from './styles';

const HomeView: FunctionComponent<IHomeViewProps> = (props: IHomeViewProps) => {
  const { name } = props;

  return (
    <HomeContainer data-testid="home-view">
      <h1>{name}</h1>
    </HomeContainer>
  );
};

export default HomeView;
