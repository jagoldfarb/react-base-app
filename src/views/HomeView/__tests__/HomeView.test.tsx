import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import HomeView from '..';

const defaultProps = {
  name: 'Home',
};

describe('Home View', () => {
  it('Should render the home view', () => {
    const { asFragment, getByTestId, getByText } = render(
      <MemoryRouter>
        <HomeView {...defaultProps} />
      </MemoryRouter>,
    );
    expect(getByTestId('home-view')).toBeDefined();
    expect(getByText('Home')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
