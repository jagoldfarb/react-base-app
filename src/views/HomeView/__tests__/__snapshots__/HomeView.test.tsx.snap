// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home View Should render the home view 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ fVMqjX"
    data-testid="home-view"
  >
    <h1>
      Home
    </h1>
  </div>
</DocumentFragment>
`;
